File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.java
class Solution {
    public int solution(String t, String p) {
        
        int len = p.length();
        int answer = 0;
        
        for(int i=0; i<t.length()-len+1; i++){
            if(t.charAt(i)-'0' > p.charAt(0)-'0') continue;
            String str = t.substring(i, i+len);
            long num = Long.parseLong(str);
            long psize = Long.parseLong(p);
            if(num <= psize) answer++;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/147355. 크기가 작은 부분문자열/크기가 작은 부분문자열.js
function solution(t, p) {
    
    var len = p.length;
    var answer = 0;
    
    for(let i=0; i<t.length-len+1; i++){
        var str = t.substring(i, i+len);
        if(Number(str) <= Number(p)) answer++;
    }
    
    return answer;
}